org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/GradleBuild.java

```diff
@@ -70,9 +70,59 @@
      * inputs/outputs/cancellation/etc.
      *
      * <p>
-     * The following sections show examples how can this API be used.
-     * TODO (donat) explain common use-cases: task execution, test execution, and model loading
-     * TODO (donat) document how to load custom models (here or link to external documentation)
+     * The following sections show examples how this API can be used.
+     *
+     * <p>
+     * <i>1. Load models</i>
+     *
+     * <p>
+     * <pre><code>
+     *     IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject("project-name");
+     *     GradleBuild build = GradleCore.getWorkspace().getBuild(project).get();
+     *
+     *     GradleProject model = build.withConnection(connection -> connection.getModel(GradleProject.class), monitor);
+     *     System.out.println(model.getBuildDirectory());
+     * </code></pre>
+     * Clients can load the <a href="https://docs.gradle.org/current/javadoc/org/gradle/tooling/ProjectConnection.html#model-java.lang.Class-"> default models</a>.
+     * Also, custom model loading possible with using {@code --init-script}:
```

Comment:
You can load custom models provided by Gradle plugins or even inject your own with an init script:

Reply:
Fixed.